app/data/views.py

```diff
@@ -29,6 +32,43 @@ def list(self, request, *args, **kwargs):
         return response
 
     @list_route(methods=['get'])
+    def stepwise(self, request):
+        """Return an aggregation counts the occurrence of events per week (per year) between
+        two bounding datetimes
+        """
+        # We'll need to have minimum and maximum dates specified to properly construct our SQL
+        try:
+            start_date = parse_date(request.GET['occurred_min'])
+            end_date = parse_date(request.GET['occurred_max'])
+        except KeyError:
+            return Response(status=HTTP_400_BAD_REQUEST)
+
+        # The min year can't be after or more than 2000 years before the max year
+        year_distance = end_date.year - start_date.year
+        if year_distance < 0 or year_distance > 2000:
+            return Response(status=HTTP_400_BAD_REQUEST)
+
+        qryset = self.get_queryset()
```

Comment:
I'm not sure the reduction in characters is worth the reduction in readability here; maybe use either `qset` or `queryset`.